### PR TITLE
building-blocks-article

### PR DESCRIPTION
**Pull request by robertdhayanturner**
## Checklist before requesting a review
- [ ] I have followed the [contribution guidelines](https://github.com/superlinked/VectorHub/tree/main/docs/tools/vdb_table)
- [ ] I have provided a reference for each attribute for which I'm adding a "support" claim.

### CODE REVIEW

File: docs/articles/basic_building_blocks.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+1. what input types the Space permits - e.g., text, timestamp, numeric, categorical
+2. whether the Space represents similarity (e.g, TextSimilaritySpace) or scale (e.g., numeric space)
+Which Space/s fit your use case depends on both these dimensions - your input type and what you need to represent about your data.
+
+You use different Spaces for different data types. For example, TextSimilaritySpace and CategoricalSimilaritySpace can take String as an input, RecencySpace can take Timestamp, NumberSpace can take Int/Float, and so on. Each Space captures a different, relevant piece of information (e.g., title, review count, etc.) about an entity. This lets you weight each Space according to which attributes are relevant to your use case - before you concatenate all your Spaces into a single multimodal vector among others in the queryable vector space.
+
+By prioritizing the creation of smarter vectors up front - and only then creating the index - we can achieve better quality retrieval, without costly and time-consuming reranking and general post-processing work.
+
+```python
+relevance_space = TextSimilaritySpace(text=paragraph.body, model="all-MiniLM-L6-v2")
```
Comment (morkapronczay) [inline docs/articles/basic_building_blocks.md:44]:
```suggestion
relevance_space = TextSimilaritySpace(text=paragraph.body, model="Snowflake/snowflake-arctic-embed-s")
```

Comment (morkapronczay) [reply docs/articles/basic_building_blocks.md:44]: let's use a more cool model :)

File: docs/articles/basic_building_blocks.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Superlinked’s Index module components enable you to group Spaces into indices that make your queries more efficient.
+
+```python
+paragraph_index = Index(relevance_space)
+```
+
+## Executing your query to your chosen endpoints
+
+Before running your code, you need to structure your query using the following arguments:
+- Query: defines the index you want it to search
```
Comment (morkapronczay) [inline docs/articles/basic_building_blocks.md:58]: let's link feature notebooks here?

also it defines the index and you can also add Params here.
param feature nb: https://github.com/superlinked/superlinked/blob/main/notebook/feature/dynamic_parameters.ipynb

File: docs/articles/basic_building_blocks.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+- .find: tells it what to look for
+- .similar: tells it how to identify relevant results
```
Comment (morkapronczay) [inline docs/articles/basic_building_blocks.md:60]: let's add: .with_vector: search with an embedded vector of an element of your data, feature nb: https://github.com/superlinked/superlinked/blob/main/notebook/feature/query_by_object.ipynb

feature nb to .similar: see for example this one, also covers find, and the topic of the whole article 
https://github.com/superlinked/superlinked/blob/main/notebook/feature/basic_building_blocks.ipynb

File: docs/articles/basic_building_blocks.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+Note that you can wait to fill out the specific Params until later.
```
Comment (morkapronczay) [inline docs/articles/basic_building_blocks.md:62]: feature nb to show this: https://github.com/superlinked/superlinked/blob/main/notebook/feature/query_time_weights.ipynb

**Review by morkapronczay — APPROVED** [review]

File: docs/articles/basic_building_blocks.md (hunk 1 above)
Comment (robertdhayanturner) [reply docs/articles/basic_building_blocks.md:44]: Should we actually use a different model here?

File: docs/articles/basic_building_blocks.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+```python
+query = (
+    Query(paragraph_index)
+    .find(paragraph)
+    .similar(relevance_space.text, Param("query_text"))
+)
```
Comment (robertdhayanturner) [inline docs/articles/basic_building_blocks.md:69]: @morkapronczay adding .with_vector above means altering the code here too. What's the addition, besides .with_vector...?

Comment (morkapronczay) [reply docs/articles/basic_building_blocks.md:69]: I see. I wasn't sure that the code and the text should be aligned, just though we list possible clauses in the query. You can probably disregard my comment then, or just put a (there is also .with_vector) in the text where I commented

File: docs/articles/basic_building_blocks.md (hunk 1 above)
Comment (ClaireSuperlinked) [reply docs/articles/basic_building_blocks.md]: Hey guys - as an FYI the BB articles in the "3 articles for Superlinked" google doc are part of the documentation project not part of VectorHub so should not be published on VectorHub

**Pull request closed without merging**